docs/spec.md

```diff
@@ -48,135 +86,158 @@ Here is a WSN defining itself:
                  "(" expression ")" |
                  "{" expression "}" .
     identifier = LETTER { LETTER } .
-    literal    = """ CHARACTER { CHARACTER } """ .
+    literal    = """ CHARACTER { CHARACTER } """ . /* """" is a literal `"` */
     LETTER     = "a" … "z" | "A" … "Z" | "_" .
     CHARACTER  = /* an arbitrary Unicode code point */ .
 
-By convention, upper case production names identify _terminal tokens_.
-Terminal tokens are the leaves in the grammar that cannot be expanded
-further. Lower case production names identify _non-terminals_, which
-are production names that may be expanded further. Lexical tokens are
-enclosed in double quotes `""`. Comments are fenced by `/* … */`.
+_Terminals_ are the leaves in the grammar that cannot be expanded
+further. By convention, terminals are identified by production names
+in uppercase.
 
-There are two special fencing tokens in evy's grammar related to
+_Non-terminals_, on the other hand, can be expanded into other
+productions. This means that they can be replaced by a more complex
+expression. By convention, non-terminals are identified by production
+names in lowercase.
+
+_Literals_ or lexical tokens are enclosed in double quotes `""`.
+Comments are fenced by `/*` … `*/`.
+
+There are two special fencing tokens in Evy's grammar related to
 horizontal whitespace, `<-` … `->` and `<+` … `+>`. `<-` … `->` means
 no horizontal whitespace is allowed between the terminals of the
 enclosed expression, e.g. `3+5` inside `<-` … `->` is allowed, but
 `3 + 5` is not. The fencing tokens `<+` … `+>` are the default and mean
 horizontal whitespace is allowed (again) between terminals.
 
-See section [whitespace](#whitespace) for further details.
+See the section on [whitespace](#whitespace) for further details.
+
+[WSN]: https://en.wikipedia.org/wiki/Wirth_syntax_notation
+[EBNF]: https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form
+[Go Programming Language Specification]: https://go.dev/ref/spec
+
+### Evy syntax grammar
 
-## Evy syntax grammar
+Evy source code is UTF-8 encoded, which means that it can contain any
+Unicode character. The NUL character `U+0000` is not allowed, as it is
+a special character that is used during compilation.
 
-The `evy` source code is UTF-8 encoded. The NUL character `U+0000` is
-not allowed.
+The `WS` abbreviation in the grammar comments below refers to horizontal
+whitespace, which is any combination of spaces and tabs. The following
+listing contains the complete syntax grammar for Evy.
 
-    program    = { statement | func | event_handler | NL } .
-    statements = statement { statement } .
+    program    = { statement | func | event_handler | nl } eof .
     statement  = typed_decl_stmt | inferred_decl_stmt |
                  assign_stmt |
                  func_call_stmt |
-                 return_stmt | break_stmt |
```

Comment:
I dont think you can remove these (return/break) from `statement`. The definition of `loop_statement` and `func_statement` which includes the corresponding `break` and `return` statements only allow those statements at the top-level of the block and not nested in an `if` for example inside a function or loop.

e.g. this is invalid according to the grammar:

```
func foo
    for i := range 5
        if i > 3
            break // invalid according to grammar - break not in `block_statements`
       else if i > 4
            return // invalid according to grammar - return not in `block_statements`
        end
    end
end
```

Reply:
good point!

Reply:
ok, I think I fixed it up. thank you for pointing this out!
I think I will wait for you to look it over one more time. this stuff is tricky.
thank you!